aces/Loris
# [Imaging Uploader] Adds validation checks in Upload tab and displays errors on form

### Brief summary of changes

This pull request adds validation checks in the Upload tab of Imaging Uploader.

Users can now only upload images if:

1. The candidate exists in the database
2. The PSCID is valid for the CandID
3. The Visit label is valid for the CandID
4. The candidate is active in the database
5. The session is active in the database
6. The file is of type .tgz or tar.gz or .zip. 
7. The file does not exceed 2M
8. The file name matches the PSCID_CandID_VisitLabel convention (once CandID, PSCID, and Visit Label are validated)

If there are validation errors, a swal window with "Submission error" will appear and the errors will be displayed under relevant form elements in the Upload form.

### This PR improves LORIS by...

- Ensuring the user submits the correct values when uploading images
- Ensuring the user uploads a file with the proper extension, size, and name
- Not taking users to a new page when there is a validation error
- Clarifying errors and displaying error messages on the form so the user can easily fix them
- Fixing small bugs

### This resolves issue...

- Redmine Task #14006: https://redmine.cbrain.mcgill.ca/issues/14006
- Redmine Task #14355: https://redmine.cbrain.mcgill.ca/issues/14355
- Additional issues were discussed in the comment section of this PR
The target branch is minor.

### To test this change...

To test this, pick a candidate in the database.

Go through the list of validations above and try to upload a file.
See if the proper errors are returned on the form.

To test validation checks for CandID, PSCID, and Visit Label,
Set Phantom Scans to "No" and try to upload a .zip file for each scenario:

1. A CandID that does not exist in the 'candidate' database
2. A PSCID that is invalid for a specific CandID in the 'candidate' database
3. A Visit Label that is invalid for a specific CandID in the 'session' database
4. A candidate that has Active set to 'N' in the 'candidate' database
5. A session that has Active set to 'N' in the 'session' database

_e.g. SQL Commands
Setting candidate associated with CandID X as inactive
UPDATE candidate SET Active = 'N' WHERE CandID = *X*;_

_Setting session associated with CandID X and Visit Label Y as inactive
UPDATE session SET Active = 'N' WHERE CandID = *X* AND Visit_label = *'Y'*;_

To test validation checks for File to Upload,
Set Phantom Scans to "No" and try to upload a file for each scenario:

6. A file that is not of type .tgz or tar.gz or .zip. 
7. A file that exceeds 2M
8. A file that that does not match the PSCID_CandID_VisitLabel convention (once CandID, PSCID, and Visit Label are validated)

You should also test **a combination of errors** or **no errors**
